packages/provider-mongo/src/example-types/facet.js

```diff
@@ -41,30 +41,33 @@ let getMatchesForMultipleKeywords = (list, optionsFilter) => ({
         list
       ),
     }),
-    _.words(optionsFilter)
+    filterWords
   ),
 })
 
-let setMatchOperators = (list, node) =>
-  list.length > 1
-    ? getMatchesForMultipleKeywords(list, node.optionsFilter)
+let setMatchOperators = (node) => {
+  const list = getSearchableKeysList(node)
+  const filterWords = _.compact(_.split(/\s+/, node.optionsFilter))
+  return list.length > 1
+    ? getMatchesForMultipleKeywords(list, filterWords)
     : {
-        $and: _.flow(
-          _.words,
-          _.map((option) => ({
+        $and: _.map(
+          (option) => ({
             [_.first(list)]: {
               $options: 'i',
               $regex: option,
             },
-          }))
-        )(node.optionsFilter),
+          }),
+          filterWords
+        ),
       }
+}
 
-let mapKeywordFilters = (node) =>
-  node.optionsFilter &&
-  _.flow(getSearchableKeysList, (list) => ({
-    $match: setMatchOperators(list, node),
-  }))(node)
+let mapKeywordFilters = (node) => [
+  // Cast field we're searching on to string to enable regex matches on it.
+  { $addFields: { _strfield: { $toString: '$_id' } } },
```

Comment:
We might want a better name than `_strfield` that communicates that it's a string version of `_id`, e.g. `_idString`

Reply:
The reason I didn't reference `id` in it is because it's not necessarily one. It's called `_id` because that's what we grouped on: https://github.com/smartprocure/contexture/blob/0c25aca67dbe4ffedf5d1aab23b5d92805bebbd2/packages/provider-mongo/src/example-types/facet.js#L152

We could name it `_nodeFieldStr` or something

Reply:
Works for me